packages/adminrouter/extra/src/lib/cache.lua

```diff
@@ -11,8 +11,18 @@ local util = require "util"
 --
 -- CACHE_FIRST_POLL_DELAY << CACHE_EXPIRATION < CACHE_POLL_PERIOD < CACHE_MAX_AGE_SOFT_LIMIT < CACHE_MAX_AGE_HARD_LIMIT
 --
--- CACHE_BACKEND_REQUEST_TIMEOUT << CACHE_REFRESH_LOCK_TIMEOUT
---
+-- There are 3 requests (2xMarathon + Mesos) made to upstream components.
+-- The cache should be kept locked for the whole time until
+-- the responses are received from all the components. Therefore,
+-- 3 * (CACHE_BACKEND_REQUEST_TIMEOUT + 2) <= CACHE_REFRESH_LOCK_TIMEOUT
+-- On the other hand, the documentation
+-- (https://github.com/openresty/lua-resty-lock#new) says that the
+-- CACHE_REFRESH_LOCK_TIMEOUT should not exceed the expiration time, which
+-- is equal to 3 * (CACHE_BACKEND_REQUEST_TIMEOUT + 2). Taking into accounts
```

Comment:
> which is equal to 3 * (CACHE_BACKEND_REQUEST_TIMEOUT + 2).

Why is that? Is that our choice?

Reply:
Unlocking the lock through a timeout is dangerous. We need to be sure that when this happens, the locked code is not still running. The locked code performs 3 HTTP requests, and each request has a timeout of `CACHE_BACKEND_REQUEST_TIMEOUT`.  Hence, the lock timeout must be at least 3 times as long. I'm not even particularly happy with the arbitrary 6 seconds added to account for all the other (non-HTTP request) locked code, but it seems reasonable to assume this will usually be safe.